app/models/pipeline_run_stage.rb

```diff
@@ -160,129 +161,41 @@ def log_url
   end
 
   ########### STAGE SPECIFIC FUNCTIONS BELOW ############
-  def prepare_dag(dag_name, attribute_dict, key_s3_params = nil)
+  def job_commands(dag_name, key_s3_params = nil)
     sample = pipeline_run.sample
     dag_s3 = "#{sample.sample_output_s3_path}/#{dag_name}.json"
-    attribute_dict[:bucket] = SAMPLES_BUCKET_NAME
-    dag = DagGenerator.new("app/lib/dags/#{dag_name}.json.erb",
-                           sample.project_id,
-                           sample.id,
-                           sample.host_genome_name.downcase,
-                           attribute_dict,
-                           pipeline_run.parse_dag_vars)
-    self.dag_json = dag.render
+    upload_dag_json(dag_json, dag_s3)
+
     copy_done_file = "echo done | aws s3 cp - #{sample.sample_output_s3_path}/\\$AWS_BATCH_JOB_ID.#{JOB_SUCCEEDED_FILE_SUFFIX}"
-    upload_dag_json_and_return_job_command(dag_json, dag_s3, dag_name, key_s3_params, copy_done_file)
+    dag_job_commands(dag_s3, dag_name, key_s3_params, copy_done_file)
```

Comment:
I suggest splitting this method in two:
    `upload_dag_json` and `dag_job_commands`, so they can be more intuitive. You can invoke both here in sequence.